### DISCUSSION TITLE
please add the Lua for "DSM tools" allowing Spektrum forward programming advanced telemetry

### DISCUSSION THREAD

**Discussion by Crunch69** [FrSkyRC/ETHOS-Feedback-Community#4164] [General]
please add the Lua for "DSM tools" Spektrum forward programming advanced telemetry.  attached is the lua for edgetx/otx please port it over to ETHOS
this will allow full control and programming for Spektrum smart ESC, RX and Telemetry
[DSMTools-main.zip](https://github.com/user-attachments/files/15856293/DSMTools-main.zip)

**Comment by Ethosfanbest** [comment]
I am really hoping this is taken seriously!

This LUA is invaluable…

I have at least five helicopters that would become so much easier to deal with in ETHOS…

I am loving this new FRSKY choice!

Thank you!

**Comment by 3Dholic** [comment]
Would be great...

**Comment by MRJordan7** [comment]
Would love to see this ported to ethos!!!!!

**Comment by toganalper** [comment]
Any update please give us a lua for Spectrum forward programming,

**Comment by bsongis-frsky** [comment]
Yes it will be done. But we are really busy with all the requests!

**Comment by oSPANNERo** [comment]
Looks like @frankiearzu has released a EthOS 1.6.0 compatible version (assuming "unofficially" but not certain):

https://github.com/frankiearzu/DSMTools/tree/main/DSM%20Tools%20Suite

**Comment by bsongis-frsky** [comment]
Indeed!
